File: 22 - Connect with JS/c. Geolocation/wasm-v7/App.razor.cs
// European Union Public License version 1.2
// Copyright © 2023 Rick Beerendonk

using System.Runtime.InteropServices.JavaScript;
using System.Runtime.Versioning;

namespace GeolocationDemo;

public partial class App
{
    [JSImport("getCurrentPosition", "App")]
    [return: JSMarshalAs<JSType.Promise<JSType.Object>>()]
    internal static partial Task<JSObject> GetCurrentPosition();
}

File: 22 - Connect with JS/c. Geolocation/wasm-v7/App.razor.js
// European Union Public License version 1.2
// Copyright © 2023 Rick Beerendonk

export async function getCurrentPosition() {
  const pos = await new Promise((resolve, reject) => {
    navigator.geolocation.getCurrentPosition(resolve, reject);
  });

  return {
    longitude: pos.coords.longitude,
    latitude: pos.coords.latitude,
  };
}
